### ISSUE TITLE
[SC2067] `find -ok` is not checked

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3332]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2067
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
find . -maxdepth 1 -exec echo {}  # SC2067
find . -maxdepth 1 -ok echo {}
find . -maxdepth 1 -ok echo {} +
find . -maxdepth 1 -okdir echo {} ';'
```

`-maxdepth` is not defined in POSIX but used so that find exits immediately.

#### Here's what shellcheck currently says:

```
myscript:2:31: error: Missing ';' or + terminating -exec. You can't use |/||/&&, and ';' has to be a separate, quoted argument. [SC2067]
```

#### Here's what I wanted or expected to see:

POSIX defines only `-ok ... ';'`, which is similar to `-exec ... ';'` but it request affirmation of the invocation.

```console
$ find /usr/bin/find -ok echo {} \;
< echo ... /usr/bin/find > ? y
/usr/bin/find
```

GNU and BSD implement `-okdir ... {} ';'` which corresponds to `-execdir ... {} ';'`; BSD implements `+` variants (multiple files at once) while GNU doesn&rsquo;t as of 4.10.0.

| predicate | terminator | POSIX | GNU | BSD |
|:--|--:|:-:|:-:|:-:|
| `-ok`    | `;` | Y | Y | Y |
| `-ok`    | `+` | N | N | Y |
| `-okdir` | `;` | N | Y | Y |
| `-okdir` | `+` | N | N | Y |

```diff
 myscript:2:31: error: Missing ';' or + terminating -exec. You can't use |/||/&&, and ';' has to be a separate, quoted argument. [SC2067]
+myscript:3:29: error: Missing ';' terminating -exec. You can't use |/||/&&, and ';' has to be a separate, quoted argument. [SC2067]
+myscript:4:29: warning: 'find ... -ok ... +' is not defined in POSIX. [SCXXXX]
+myscript:5:20: warning: 'find ... -okdir ...' is not defined in POSIX. [SCXXXX]
```
